hazelcast/src/main/java/com/hazelcast/map/impl/operation/BaseRemoveOperation.java

```diff
@@ -45,6 +54,14 @@ protected void afterRunInternal() {
         super.afterRunInternal();
     }
 
+    @Override
+    public void applyState(State state) {
+        super.applyState(state);
+
+        this.dataKey = state.getKey();
+        this.dataOldValue = mapServiceContext.toData(state.getOldValue());
```

Comment:
Currently, we're performing this `this.dataOldValue = mapServiceContext.toData(state.getOldValue());` serialization call twice both in the child classes and here. This situation also applies to the `BasePutOperation` and its child classes.

Reply:
good catch, fixed, it should be one time conversion now.